src/main/java/com/autonomic/tmc/auth/ClientCredentialsTokenSupplier.java

```diff
@@ -73,100 +57,81 @@
-     * requested automatically. This method will always return a valid token.
+     * This method responds with a valid String representation of the Bearer token.  If a token
+     * becomes expired, a new valid token will be requested automatically. This method will always
+     * return a valid token.
      *
      * @return String representation of Bearer token.
-     * @throws AuthenticationCommunicationException Thrown when an unexpected condition is encountered while making the client credentials
-     * grant POST
-     * @throws AuthenticationFailedException Thrown when the credentials that were provided are expressly rejected.
+     * @throws AuthenticationCommunicationException Thrown when an unexpected condition is
+     *                                              encountered while making the client credentials
+     *                                              grant POST
+     * @throws AuthenticationFailedException        Thrown when the credentials that were provided
+     *                                              are expressly rejected.
      */
     @Override
     public synchronized String get() {
         if (token != null && !token.isExpired()) {
+            log.debug("Cached token found.");
             return token.getValue();
         }
         resetToken();
         return token.getValue();
     }
 
     Token getExistingToken() {
-        return this.token;
+        return token;
     }
 
     private void resetToken() {
         // if not already reset
         if (token == null || token.isExpired()) {
+            log.debug("No cached token. Retrieving a new token.");
             token = authenticate();
         }
     }
 
     private Token authenticate() {
-
         TokenResponse response = executeTokenRequest();
         AccessToken accessToken = processTokenResponse(response);
 
-        return Token.builder()
-            .value(accessToken.getValue())
-            .expiration(Instant.now().plus(accessToken.getLifetime(), ChronoUnit.SECONDS))
-            .build();
+        return baseSupplier.buildToken(accessToken);
     }
 
     private AccessToken processTokenResponse(TokenResponse response) {
-
-        if (!response.indicatesSuccess()) {
-            HTTPResponse httpResponse = response.toHTTPResponse();
-            int responseCode = httpResponse.getStatusCode();
-            if (responseCode == 401 || responseCode == 400) {
-                throw new AuthenticationFailedException(String
-                    .format("Authorization failed for user [%s] at tokenUrl [%s]",
-                        this.clientId, this.tokenUrl));
+        try {
+            return baseSupplier.processTokenResponse(response);
+        } catch (SdkServiceException e) {
+            int statusCode = e.getStatusCode();
+            if (statusCode == 401 || statusCode == 400) {
+                throw new AuthenticationFailedException(e.getMessage());
```

Comment:
This is interesting to me.  Do we want to have a SDKException and then through a specific exception?

Reply:
This class is being deleted.